editor/src/components/navigator/navigator-item/run-in-shard-2-component-picker-context-menu.spec.browser2.tsx

```diff
@@ -1602,5 +1602,132 @@ export const Column = () => (
         expectedOutput,
       )
     })
+
+    it('Works when replacing the root element of a component', async () => {
+      const editor = await renderTestEditorWithCode(
+        `import * as React from 'react'
+      import {
+        Scene,
+        Storyboard,
+        Ellipse,
+      } from 'utopia-api'
+      
+      const Playground = ({ children, style }) => (
+        <div data-uid='pg-root' style={style}>
+          {children}
+        </div>
+      )
+      
+      export var storyboard = (
+        <Storyboard data-uid='sb'>
+          <Scene
+            id='playground-scene'
+            commentId='playground-scene'
+            style={{
+              width: 700,
+              height: 759,
+              position: 'absolute',
+              left: 212,
+              top: 128,
+            }}
+            data-label='Playground'
+            data-uid='scene'
+          >
+            <Playground style={{}} data-uid='pg'>
+              <Ellipse
+                style={{
+                  backgroundColor: '#aaaaaa33',
+                  width: 100,
+                  height: 100,
+                }}
+                data-uid='aaa'
+              />
+              <Ellipse
+                style={{
+                  backgroundColor: '#aaaaaa33',
+                  width: 100,
+                  height: 100,
+                }}
+                data-uid='aag'
+              />
+              <Ellipse
+                style={{
+                  backgroundColor: '#aaaaaa33',
+                  width: 100,
+                  height: 100,
+                }}
+                data-uid='aai'
+              />
+            </Playground>
+          </Scene>
+        </Storyboard>
+      )
+      `,
+        'await-first-dom-report',
+      )
+      const targetPath = EP.fromString('sb/scene/pg:pg-root')
+      await selectComponentsForTest(editor, [targetPath])
+
+      const replaceButton = editor.renderedDOM.getByTestId(
+        ReplaceElementButtonTestId(targetPath, null),
+      )
+      await mouseClickAtPoint(replaceButton, { x: 2, y: 2 })
+
+      const menuButton = await waitFor(() => editor.renderedDOM.getByText('FlexCol'))
+      await mouseClickAtPoint(menuButton, { x: 3, y: 3 })
+
+      await editor.getDispatchFollowUpActionsFinished()
+
+      expect(getPrintedUiJsCodeWithoutUIDs(editor.getEditorState(), StoryboardFilePath))
+        .toEqual(`import * as React from 'react'
+import { Scene, Storyboard, Ellipse } from 'utopia-api'
+import { FlexCol } from 'utopia-api'
+
+const Playground = ({ children, style }) => (
+  <FlexCol style={{ position: 'absolute' }} />
```

Comment:
can you make a test that demonstrates that it can also retain its children?

Reply:
added it in https://github.com/concrete-utopia/utopia/pull/5681/commits/3e963227b4957eb5cdd73b8c9c6df21d94e65f32